velox/type/tests/VariantTest.cpp

```diff
@@ -142,10 +143,97 @@ TEST(Variant, longDecimal) {
       v.value<TypeKind::LONG_DECIMAL>(), u2.value<TypeKind::LONG_DECIMAL>());
 }
 
-TEST(Variant, longDecimalNull) {
+TEST(VariantTest, longDecimalNull) {
   variant null = variant::longDecimal(std::nullopt, DECIMAL(20, 5));
   EXPECT_TRUE(null.isNull());
   EXPECT_EQ(null.toJson(), "null");
   EXPECT_EQ(*null.inferType(), *DECIMAL(20, 5));
   EXPECT_THROW(variant::null(TypeKind::LONG_DECIMAL), VeloxException);
 }
+
+TEST(VariantTest, equalsWithEpsilonDouble) {
+  std::vector<int64_t> data = {
+      -6524373357247204968,
+      -1459602477200235160,
+      -5427507077629018454,
+      -6362318851342815124,
+      -6567761115475435067,
+      9193194088128540374,
+      -7862838580565801772,
+      -7650459730033994045,
+      327870505158904254,
+  };
+
+  double sum1 = std::accumulate(data.begin(), data.end(), 0.0);
+
+  double sumEven = 0;
+  double sumOdd = 0;
+  for (auto i = 0; i < data.size(); i++) {
+    if (i % 2 == 0) {
+      sumEven += data[i];
+    } else {
+      sumOdd += data[i];
+    }
+  }
+
+  double sum2 = sumOdd + sumEven;
+
+  ASSERT_NE(sum1, sum2);
+  ASSERT_DOUBLE_EQ(sum1, sum2);
+  ASSERT_TRUE(variant(sum1).equalsWithEpsilon(variant(sum2)));
+
+  // Add up all numbers by one. Make sure the result is not equal to sum1.
```

Comment:
Is the comment consistent with the implementation?

Reply:
Good catch. I think I have a typo. It should say "all numbers BUT one". Let me fix.